cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -260,6 +261,15 @@ protected boolean doInstall(boolean silent) {
 
     // check if we already have this version installed (linked) locally in IDE_HOME/software
     VersionIdentifier installedVersion = getInstalledVersion();
+
+    UrlSecurityFile securityFile = this.context.getUrls().getEdition(this.tool, this.getEdition()).getSecurityFile();
+    // I do not want to use the installed version here, as I want to warn the user whether the tool is installed or not.
```

Comment:
I am not so sure if that makes sense.
Actually when we start downloading a specific version we should raise the warning and potentially even ask the user for confirmation. Therefore this should go somewhere into `ToolRepository.download`.

Reply:
Okey, I will move it to `ToolRepository.download` and prompt the user with a question if the version has a CVE. Then when running the tool, should there be also a warning, or no warning, or also a prompt?

Reply:
Okey, I have moved it to `doInstall()` of `LocalToolCommandlet` and `GloballToolCommandlet`, which is in my IMHO the best place for it.